packages/contracts-bedrock/contracts/universal/CrossDomainMessenger.sol

```diff
@@ -369,13 +369,16 @@ abstract contract CrossDomainMessenger is
      *
      * @return Amount of gas required to guarantee message receipt.
      */
-    function baseGas(bytes calldata _message, uint32 _minGasLimit) public pure returns (uint32) {
+    function baseGas(bytes calldata _message, uint32 _minGasLimit) public pure returns (uint64) {
```

Comment:
Note: 
The inputs are unchanged, but the internal math and return value are now `uint64`. Everywhere its used is already accepting a `uint64` value though: 

ie. in `_sendMessage`: 

https://github.com/ethereum-optimism/optimism/blob/4bc75740ad830b51cf0ad61bce63e4bb37158736/packages/contracts-bedrock/contracts/universal/CrossDomainMessenger.sol#L410

Which is implemented on L1 as: 
https://github.com/ethereum-optimism/optimism/blob/4bc75740ad830b51cf0ad61bce63e4bb37158736/packages/contracts-bedrock/contracts/L1/L1CrossDomainMessenger.sol#L45-L52

and on L2 as:

https://github.com/ethereum-optimism/optimism/blob/4bc75740ad830b51cf0ad61bce63e4bb37158736/packages/contracts-bedrock/contracts/L2/L2CrossDomainMessenger.sol#L51-L60

Reply:
A more general question though is "what happens when either of these functions get a gas limit greater than the block gas limit". 
I think that requires an iTest.

Reply:
The reason for uint32 was because `sendMessage` takes uint32 as input: https://github.com/ethereum-optimism/optimism/blob/e59b44965256a0d3a20166dabccc523a74a10d14/packages/contracts-bedrock/contracts/universal/CrossDomainMessenger.sol#L219-L223

Reply:
This is a holdover from the old `sendMessage` interface: https://github.com/ethereum-optimism/optimism/blob/e59b44965256a0d3a20166dabccc523a74a10d14/packages/contracts/contracts/L1/messaging/L1CrossDomainMessenger.sol#L137-L141

Reply:
To be honest I don't know why `_gasLimit` was changed from a `uint32` to a `uint64`. `uint32` was sufficient.

Reply:
If anything my preference would actually be to go in the other direction and change the new interface to use `uint32` instead of `uint64` and keep everything more consistent.

Reply:
There's no point in having `uint64` in `_sendMessage` if the public `sendMessage` only accepts `uint32`.

Reply:
The reason why I think this change makes sense is that the entire set of `uint32`s is not valid inputs to `sendMessage` because `_baseGas` will pad `_gasLimit` to a larger value. By making it a `uint64`, all `uint32`s are valid inputs. Users should never be getting `Panic(0x11)` overflow errors back by setting `_gasLimit` too high

Reply:
@tynes where does the `Panic(0x11)` come from?

Reply:
> By making it a `uint64`, all `uint32s` are valid inputs. Users should never be getting `Panic(0x11)` overflow errors back by setting _gasLimit too high

In practice, the amount required to cause an overflow is going to be very close to `uint32` (1000/1016ths), so about 4 Billion gas.


@tynes how do you feel about [this approach](https://github.com/ethereum-optimism/optimism/blob/6b75037ca3545b8c60eef252badfb43d66a6cb59/packages/contracts-bedrock/contracts/universal/CrossDomainMessenger.sol#L376-L386) which would throw an `Arithmetic: overflow/underflow` error on the downcast step?

If we want a more descriptive message, we could add an explicit check, something like:
`require(output < type(uint32).max, "The amount of gas supplied would lead to an excessively large gas limit")`

Reply:
> @tynes where does the `Panic(0x11)` come from?

`solc` returns this on overflow/underflow

Reply:
resolving this lengthy thread in favor of discussion about the types elsewere.